syntax = "proto3";
package os.logger;

import "Common.proto";

message TargetDetection {
  Time   t = 1;
  uint32 frame_id = 2;

  // --- Detection validity and IDs ---
  bool   is_target_detected = 10;     // was: target_valid
  int32  detection_id       = 11;     // was: target_detection_id
  int32  track_id           = 12;     // was: target_track_id
  int32  class_id           = 13;     // was: detection_class
  float  confidence         = 14;     // was: target_detection_conf

  // --- Bounding box geometry (pixel units) ---
  float bbox_center_offset_x_px_raw     = 20;    // was: target_cntr_offset_x
  float bbox_center_offset_y_px_raw     = 21;    // was: target_cntr_offset_y
  float bbox_height_px       = 22;    // was: target_height
  float bbox_width_px        = 23;    // was: target_width
  float bbox_aspect_ratio    = 24;    // was: target_aspect
  float bbox_left_px         = 25;    // was: target_left
  float bbox_right_px        = 26;    // was: target_right
  float bbox_top_px          = 27;    // was: target_top
  float bbox_bottom_px       = 28;    // was: target_bottom
  float bbox_center_offset_x_px_filt     = 29;    // was: target_center_x
  float bbox_center_offset_y_px_filt     = 30;    // was: target_center_y
}

message TargetState {
  Time   t = 1;
  uint32 frame_id = 2;

  // --- Camera and angular geometry ---
  float delta_angle       = 10;   // was: delta_angle
  float camera_tilt_angle       = 11;   // was: camera_tilt_angle
  float delta_distance_x_m    = 12;   // was: delta_d_x
  float delta_distance_z_m    = 13;   // was: delta_d_z

  // --- EKF-estimated target state ---
  float pos_x_est      = 20;   // was: x_target_ekf
  float pos_y_est      = 21;   // was: y_target_ekf
  float vel_x_est      = 22;   // was: vx_target_ekf
  float vel_y_est      = 23;   // was: vy_target_ekf
  float acc_x_est      = 24;   // was: ax_target_ekf
  float acc_y_est      = 25;   // was: ay_target_ekf

  // --- Post-filter measurement usability ---
  bool  is_measurement_valid  = 30;   // was: target_data_useful
}
